avbroot/src/cli/avb.rs

```diff
@@ -194,6 +194,121 @@ fn write_raw_and_update(
     Ok(raw_file)
 }
 
+/// Compute the kernel command line arguments to allow the kernel to set up the
+/// dm-verity block device without dm-init or userspace helpers. This is handled
+/// by init/do_mounts_dm.c in older Pixel devices (and ChromiumOS).
+fn compute_dm_verity_cmdline(descriptor: &HashTreeDescriptor) -> String {
+    use std::fmt::Write;
+
+    let mut result = String::new();
+
+    // Number of device mapper devices.
+    result.push_str("dm=\"1");
+    // Block device name.
+    result.push_str(" vroot");
+    // Block device UUID.
+    result.push_str(" none");
+    // Block device write mode.
+    result.push_str(" ro");
+    // Number of device mapper targets.
+    result.push_str(" 1,");
+    // Starting sector.
+    result.push('0');
+    // Sector count.
+    write!(&mut result, " {}", descriptor.image_size / 512).unwrap();
```

Comment:
Wouldn't using `unwrap()` result in the to panic if it fails? I suggest using `unwrap_or_default()`,`unwrap_or()`, or better using a `map()` instead.. 😅

Reply:
Normally yes, but formatting into a `String` is guaranteed to never fail (unless the system runs out of memory, but Rust will panic in that case anyway).